synapse/storage/database.py

```diff
@@ -1512,7 +1529,7 @@ def get_cache_dict(
 
         return cache, min_val
 
-    def simple_select_list_paginate(
+    async def simple_select_list_paginate(
```

Comment:
This method actually isn't used, only `simple_select_list_paginate_txn` is used -- do we think we should just get rid of it?

Reply:
yes!

Reply:
(this was part of the disastrous #1784 which we have been gradually reverting for the last 3.5 years...)